### DISCUSSION TITLE
How to navigate to a specific view from Dynamic Link - iOS

### DISCUSSION THREAD

**Discussion by LucasHimelfarb** [invertase/react-native-firebase#6227] [Q&A]
**MY SETUP:**

  **Firebase dependencies:**
  ```
        "@react-native-firebase/analytics": "^11.1.2",
        "@react-native-firebase/app": "^11.1.2",
        "@react-native-firebase/crashlytics": "^11.1.2",
        "@react-native-firebase/dynamic-links": "^11.5.0"
  ```
  
  **React native version:** _"react-native": "0.67.1"_
  
**SO:** _macOS Monterey 12.3.1_
**Chip:** _M1_
**Mobile device:** _Iphone 7 with iOS 15.4.1_

**PROBLEM:**
_my dynamic link has to open the application and navigate to a specific view. In android it works fine but in iOS it only opens the application, but does not navigate._

**Errors from Xcode:**
 _Deep Link Web URL query is empty
RNFBDynamicLinks: Unknown error occurred when attempting to handle a universal link: Error Domain=com.firebase.dynamicLinks Code=404 "(null)" UserInfo={message=Requested entity was not found., status=NOT_FOUND, code=404}_

**Comment by mikehardy** [comment]
These are very very old versions, and they are mismatched!

```
      "@react-native-firebase/analytics": "^11.1.2",
      "@react-native-firebase/app": "^11.1.2",
      "@react-native-firebase/crashlytics": "^11.1.2",
      "@react-native-firebase/dynamic-links": "^11.5.0"
```

https://invertase.io/blog/react-native-firebase-versioning - please update, consulting the changelog for the exact changes to make in order to handle the major version bumps, and try again? The newer version of the underlying firebase-ios-sdk 8.15.0 may just fix it for you.

If not, you might try checking the built-in react-native linking onIniitialLink to see if it contains the link, at which point you can use dynamic-links api here to resolve it in a startup service and navigate. My apps use code like that as a fallback

Hopefully that helps get things moving?

**Reply by LucasHimelfarb** [reply]
@mikehardy thanks a lot for you answer! Please look my package-lock.json:

  ```
  "node_modules/@react-native-firebase/analytics": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/analytics/-/analytics-11.5.0.tgz",
        "@react-native-firebase/app": "11.5.0"
    "node_modules/@react-native-firebase/app": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/app/-/app-11.5.0.tgz",
    "node_modules/@react-native-firebase/crashlytics": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/crashlytics/-/crashlytics-11.5.0.tgz",
        "@react-native-firebase/app": "11.5.0"
    "node_modules/@react-native-firebase/dynamic-links": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/dynamic-links/-/dynamic-links-11.5.0.tgz",
        "@react-native-firebase/app": "11.5.0"
    "node_modules/@react-native-firebase/messaging": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/messaging/-/messaging-11.5.0.tgz",
        "@react-native-firebase/app": "11.5.0"
    "@react-native-firebase/analytics": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/analytics/-/analytics-11.5.0.tgz",
    "@react-native-firebase/app": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/app/-/app-11.5.0.tgz",
    "@react-native-firebase/crashlytics": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/crashlytics/-/crashlytics-11.5.0.tgz",
    "@react-native-firebase/dynamic-links": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/dynamic-links/-/dynamic-links-11.5.0.tgz",
    "@react-native-firebase/messaging": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/messaging/-/messaging-11.5.0.tgz",
```

at first glance the versions coincide. Will it be necessary to update? I don't get any errors that break my app.

This is the documentation that I was following: https://rnfirebase.io/dynamic-links/usage

**Reply by mikehardy** [reply]
Okay, good to see that despite the mismatch spec in package.json they are resolving correctly
Will it be necessary to update? That's an existential question.

If you imagined yourself in a sort of helpdesk role, offering support for software though. And you issued bugfix releases frequently, then a software user comes in and says I have a problem but they are on an old version. What do you say? Do you attempt to troubleshoot an old version?

I don't, personally. So if you want to discuss this as a possible issue and access support: use the current version. Otherwise, I dunno - it may be possible to figure it out. I had it working on v11.5 if I recall correctly. But mobile operating systems change, and the SDKs issue updates to handle the changes and I've never tested v11.5 on current mobile operating systems :shrug:

**Reply by LucasHimelfarb** [reply]
Ok Mike I thought that the error could be something known, related to the Dynamic Links setup in iOS... Some wrong step that causes this unwanted behavior. I'll try to jump to the latest versions of Firebase and tell you how it works on iOS and Android again. Thanks a lot for you time!

**Reply by LucasHimelfarb** [reply]
@mikehardy hello mike, how are you today? Update firebase as you recommended, luckily it wasn't too difficult

```
        "@react-native-firebase/analytics": "^14.9.1",
        "@react-native-firebase/app": "^14.9.1",
        "@react-native-firebase/crashlytics": "^14.9.1",
        "@react-native-firebase/dynamic-links": "^14.9.1",
        "@react-native-firebase/messaging": "^14.9.1",
    "node_modules/@react-native-firebase/analytics": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/analytics/-/analytics-14.9.1.tgz",
        "@react-native-firebase/app": "14.9.1"
    "node_modules/@react-native-firebase/app": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/app/-/app-14.9.1.tgz",
    "node_modules/@react-native-firebase/crashlytics": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/crashlytics/-/crashlytics-14.9.1.tgz",
        "@react-native-firebase/app": "14.9.1"
    "node_modules/@react-native-firebase/dynamic-links": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/dynamic-links/-/dynamic-links-14.9.1.tgz",
        "@react-native-firebase/app": "14.9.1"
    "node_modules/@react-native-firebase/messaging": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/messaging/-/messaging-14.9.1.tgz",
        "@react-native-firebase/app": "14.9.1"
    "@react-native-firebase/analytics": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/analytics/-/analytics-14.9.1.tgz",
    "@react-native-firebase/app": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/app/-/app-14.9.1.tgz",
    "@react-native-firebase/crashlytics": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/crashlytics/-/crashlytics-14.9.1.tgz",
    "@react-native-firebase/dynamic-links": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/dynamic-links/-/dynamic-links-14.9.1.tgz",
    "@react-native-firebase/messaging": {
      "resolved": "https://registry.npmjs.org/@react-native-firebase/messaging/-/messaging-14.9.1.tgz",
```

Returning to my initial problem, I think there is something in the documentation that I have not fully understood, in Xcode I have the following error:

**RNFBDynamicLinks: Unknown error occurred when attempting to handle a universal link: Error Domain=com.firebase.dynamicLinks Code=404 "(null)" UserInfo={message=Requested entity was not found., status=NOT_FOUND, code=404}**


I am using a page.link. Could it be for this reason that iOS ignores the full path?


Thanks for all! Especially for your time and sorry for the inconvenience

**Reply by mikehardy** [reply]
Well, at least you are on current versions now, but I'm sorry that did not result in progress on your issue
Have you gone through all these steps https://rnfirebase.io/dynamic-links/usage#ios-troubleshooting to make sure that your app is associated with the link domain etc? 

Have you tried using the built-in react-native APIs to see if you get anything? https://reactnative.dev/docs/linking#getinitialurl ?

**Reply by LucasHimelfarb** [reply]
Yes, with my new link generated from Firebase I can navigate to my app store if it is not installed. I have not specifically used Linking but if I tried to obtain the link from javascript as you recommended and it worked with it: `const dynamicLinkUrl = await dynamicLinks().getInitialLink()`. From the documentation I omitted CustomLinks, maybe I am missing height in my info.plist
